src/packets.rs

```diff
@@ -232,82 +348,221 @@ impl PacketBuffer {
     ///   0000000000000000: 90 3c 7f
     ///   0000000000000000: 90 40 7f
     /// ```
-    pub fn with_data(mut self, timestamp: Timestamp, data: Vec<u8>) -> Self {
-        let data_len = data.len();
-        assert!(data_len < MAX_PACKET_DATA_LENGTH,
-                "The maximum allowed size for a packet is {}, but found {}.",
-                MAX_PACKET_DATA_LENGTH, data_len);
-
-        let additional_size = PACKET_SIZE + data_len;
-        self.data.reserve(additional_size);
-
-        let mut pkt = unsafe {
-            let total_len = self.data.len();
-            self.data.set_len(total_len + additional_size);
-            &mut *(&self.data[total_len] as *const _ as *mut MIDIPacket)
-        };
-
-        pkt.timeStamp = timestamp as MIDITimeStamp;
-        pkt.length = data_len as UInt16;
-        pkt.data[0..data_len].clone_from_slice(&data);
+    pub fn push_packet(&mut self, time: MIDITimeStamp, data: &[u8]) -> &mut Self {
```

Comment:
Does `data` represent a `Packet` or a list of MIDI bytes ? The name `push_packet` is confusing. I'd rename it to `push_data` to make this differentiation clear.

Reply:
Following the same logic I would also rename `assign_packet` and `extend_packet` for the `PacketBufferStorage`.

Reply:
`assign_packet` assigns data to a single `MIDIPacket`, `extend_packet` extends the data inside a single `MIDIPacket`, so I think that these names are okay. You are right that `push_packet` either creates a new packet or extends an existing one, so `push_data` is probably better. I will rename it.

Reply:
I now remember why I didn't call it `push_data` from the beginning: It is required that the data that is being pushed in a single call of this function actually is a valid packet and not arbitrary MIDI bytes. E.g. it is not allowed to push a NOTE ON followed by a SYSEX event in a single "packet".  Also see the [original documentation of MIDIPacketListAdd](https://github.com/phracker/MacOSX-SDKs/blob/9fc3ed0ad0345950ac25c28695b0427846eea966/MacOSX10.13.sdk/System/Library/Frameworks/CoreMIDI.framework/Versions/A/Headers/MIDIServices.h#L2146-L2150). There, a "packet" is called an "event", so we could also call this `push_event`. I think I still prefer `push_data` now, under the condition that the documentation makes it clear what is allowed as data passed to the function.